src/ops.rs

```diff
@@ -33,6 +72,60 @@ pub fn pop(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Result
     Ok(())
 }
 
+pub fn bit_scan_forward(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Result<(), VMError> {
+    if pipeline.size_override {
+        let source = vm.get_arg(pipeline.args[0].location)?.u16_exact()?;
+        let binary_string = format!("{:b}", source);
```

Comment:
Doing this via strings and format! is rather scary from a performance perspective. This shouldn't be too difficult to do manually with bitwise operations, but I think this crate might make it a lot more straight forward and maybe get some use in other areas https://crates.io/crates/bitmaps